### PR TITLE
Update to 0.11.0

### PR DESCRIPTION
**Pull request by kszucs**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by kszucs** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
 
 package:
   name: arrow-cpp
   version: {{ version }}
 
 source:
-  fn: {{ commit }}.tar.gz
-  url: https://github.com/apache/arrow/archive/{{ commit }}.tar.gz
+  fn: apache-arrow-{{ version }}.tar.gz
+  url: https://github.com/apache/arrow/archive/apache-arrow-{{ version }}.tar.gz
```
Comment (kszucs) [inline recipe/meta.yaml]: I suppose it'd be better to use the official apache archive instead.

Comment (wesm) [reply recipe/meta.yaml]: Indeed. I think it's OK to use the dist link here since this job doesn't run often

**Comment by kszucs** [comment]
@wesm please review

**Pull request merged**